tests/sys/psa_crypto_mac/test_hmac_sha384.c

```diff
@@ -73,207 +59,374 @@ static psa_status_t test_hmac_sha384(const unsigned char *key, size_t key_len,
     status = psa_import_key(&attributes, key, key_len, &key_handle);
     if (status != PSA_SUCCESS) {
         psa_destroy_key(key_handle);
-        return status;
+        TEST_FAIL("Key import failed");
     }
 
     status = psa_mac_sign_setup(&operation, key_handle, PSA_ALG_HMAC(PSA_ALG_SHA_384));
     if (status != PSA_SUCCESS) {
         psa_destroy_key(key_handle);
-        return status;
+        TEST_FAIL("MAC setup failed");
     }
 
-    status = psa_mac_update(&operation, message, message_len);
+    status = psa_mac_update(&operation, data, data_len);
     if (status != PSA_SUCCESS) {
         psa_destroy_key(key_handle);
-        return status;
+        TEST_FAIL("MAC update failed");
     }
-    size_t hmac_length;
-    status = psa_mac_sign_finish(&operation, hmac, SHA384_DIGEST_LENGTH, &hmac_length);
+
+    status = psa_mac_sign_finish(&operation, hmac, sizeof(hmac), &hmac_length);
     if (status != PSA_SUCCESS) {
         psa_destroy_key(key_handle);
-        return status;
+        TEST_FAIL("MAC finishing failed");
     }
 
     psa_destroy_key(key_handle);
 
-    return PSA_SUCCESS;
+    TEST_ASSERT_MESSAGE(hmac_length == sizeof(hmac), "HMAC result size mismatch");
+    /* Using result_len is important to allow for truncating the checked data */
+    TEST_ASSERT_MESSAGE(memcmp(hmac, result, result_len) == 0, message);
 }
 
-static void test_hashes_hmac_sha384_hash_sequence(void)
+static const uint8_t sequence_key_1[64] = {
+    0x00
+};
+
+/* We cannot directly create zero-sized arrays */
+static const uint8_t sequence_data_1[1] = {
+    0x00
+};
+
+static const uint8_t sequence_hmac_1[48] = {
+    0x6c, 0x1f, 0x2e, 0xe9, 0x38, 0xfa, 0xd2, 0xe2, 0x4b, 0xd9,
+    0x12, 0x98, 0x47, 0x43, 0x82, 0xca, 0x21, 0x8c, 0x75, 0xdb,
+    0x3d, 0x83, 0xe1, 0x14, 0xb3, 0xd4, 0x36, 0x77, 0x76, 0xd1,
+    0x4d, 0x35, 0x51, 0x28, 0x9e, 0x75, 0xe8, 0x20, 0x9c, 0xd4,
+    0xb7, 0x92, 0x30, 0x28, 0x40, 0x23, 0x4a, 0xdc
+};
+
+/* "key" */
+static const uint8_t sequence_key_2[64] = {
+    0x6b, 0x65, 0x79
+};
+
+/* "The quick brown fox jumps over the lazy dog" */
+static const uint8_t sequence_data_2[43] = {
+    0x54, 0x68, 0x65, 0x20, 0x71, 0x75, 0x69, 0x63, 0x6B, 0x20,
+    0x62, 0x72, 0x6F, 0x77, 0x6E, 0x20, 0x66, 0x6F, 0x78, 0x20,
+    0x6A, 0x75, 0x6D, 0x70, 0x73, 0x20, 0x6F, 0x76, 0x65, 0x72,
+    0x20, 0x74, 0x68, 0x65, 0x20, 0x6C, 0x61, 0x7A, 0x79, 0x20,
+    0x64, 0x6F, 0x67
+};
```

Comment:
```suggestion
static const uint8_t sequence_data_2[43] = "The quick brown fox jumps over the lazy dog";
```

should do the trick (the zero-terminated string gets truncated at the array size)

Reply:
I would rather keep the old arrax style so that the test cases have the same look.

Reply:
Okay, fine with me :)